### PR TITLE
require an -mnt suffix in maintainer name

### PR DESCRIPTION
**Pull request by dadepo**

### CODE REVIEW

**Comment by eshryane** [comment]
The syntax check is going to prevent modifying and deleting existing mntners. The extra validation should *only* be done on create. I suggest using a BusinessRuleValidator (i.e. have a look at MaintainerNameValidator as an example).

**Comment by dadepo** [comment]
> The syntax check is going to prevent modifying and deleting existing mntners. The extra validation should _only_ be done on create. I suggest using a BusinessRuleValidator (i.e. have a look at MaintainerNameValidator as an example).

Updated as suggested

File: whois-update/src/main/java/net/ripe/db/whois/update/handler/validator/maintainer/MaintainerNameValidator.java [hunk 1]
```diff
@@ -32,6 +33,9 @@
     @Override
     public void validate(final PreparedUpdate update, final UpdateContext updateContext) {
         final RpslObject updatedObject = update.getUpdatedObject();
+        if (Action.CREATE.equals(update.getAction()) && !updatedObject.getKey().toUpperCase().endsWith(MNT_NAME_SUFFIX)) {
```
Comment (eshryane) [inline whois-update/src/main/java/net/ripe/db/whois/update/handler/validator/maintainer/MaintainerNameValidator.java:36]: No need for toUpperCase(), CIString implements endsWith()

File: whois-update/src/main/java/net/ripe/db/whois/update/domain/UpdateMessages.java [hunk 2]
```diff
@@ -176,6 +176,10 @@ public static Message invalidMaintainerForOrganisationType(final CharSequence or
         return new Message(Type.ERROR, "Value '%s' can only be set by the RIPE NCC for this organisation.", orgType);
     }
 
+    public static Message invalidMaintainerName() {
+        return new Message(Type.ERROR, "Maintainer name must end with a -MNT suffix");
```
Comment (eshryane) [inline whois-update/src/main/java/net/ripe/db/whois/update/domain/UpdateMessages.java:180]: Update the message to clarify that the error is on CREATE only:
`When creating a MNTNER the name must end with an -MNT suffix`

File: whois-endtoend/src/test/groovy/net/ripe/db/whois/spec/update/UpdateMaintainerSpec.groovy [hunk 3]
```diff
@@ -304,6 +323,40 @@ class UpdateMaintainerSpec extends BaseQueryUpdateSpec {
         queryObject("-rGBT mntner CRE-MNT", "mntner", "CRE-MNT")
     }
 
+    def "create maintainer object with invalid  (without -mnt suffix) maintainer name"() {
```
Comment (eshryane) [inline whois-endtoend/src/test/groovy/net/ripe/db/whois/spec/update/UpdateMaintainerSpec.groovy:326]: Is there a successful test to create a maintainer with an -mnt suffix?

File: whois-update/src/main/java/net/ripe/db/whois/update/handler/validator/maintainer/MaintainerNameValidator.java (hunk 1 above)
Comment (dadepo) [reply whois-update/src/main/java/net/ripe/db/whois/update/handler/validator/maintainer/MaintainerNameValidator.java]: Removed

File: whois-endtoend/src/test/groovy/net/ripe/db/whois/spec/update/UpdateMaintainerSpec.groovy (hunk 3 above)
Comment (dadepo) [reply whois-endtoend/src/test/groovy/net/ripe/db/whois/spec/update/UpdateMaintainerSpec.groovy:326]: Ys, there are existing test that checks the creation of a maintainer and they used -mnt suffix, for example `create self maintained maintainer object check pw hash filtering` and `create maintainer object maintained by other mntner`

File: whois-update/src/main/java/net/ripe/db/whois/update/domain/UpdateMessages.java (hunk 2 above)
Comment (dadepo) [reply whois-update/src/main/java/net/ripe/db/whois/update/domain/UpdateMessages.java]: done

**Review by eshryane — APPROVED** [review]

**Pull request merged**